backend/wargamebackend/wargamelogic/consumers.py

```
@@ -157,82 +161,98 @@ async def handle_message(self, event):
 
     async def handle_users_join(self, data):
         """
         data: RoleInstance
         """
         user = self.scope["user"]
-        if data.get("user", {}).get("username") != user.username:
-            raise Exception(f"{user.username} tried joining as {data.get('user', {}).get('username')} in game '{self.join_code}'.")
+        user_username = data["user"]["username"]
+
+        if user_username != user.username:
+            raise Exception(f"{user.username} tried joining as {user_username} in game '{self.join_code}'.")
```

Ruff (0.12.2)

Create your own exception

(TRY002)